### ISSUE TITLE
[BUG] `no-custom-classname` doesn't work with hyphenated group names

### ISSUE THREAD

**Issue by lachieh** [francoismassart/eslint-plugin-tailwindcss#226]
**Describe the bug**
The `no-custom-classname` hits a false-positive when group names have anything other than `[a-zA-Z0-9]`

**To Reproduce**
```html
<element class="group/custom-name">...</element>
```
```sh
  1:16  warning  Classname 'group/custom-name' is not a Tailwind CSS class!  tailwindcss/no-custom-classname
```

**Expected behavior**
Should accept anything that is a valid css rule. From my investigation it looks like almost any character except space. Tailwind itself handles these classes correctly:

https://play.tailwindcss.com/pveMxjkzQb
<img width="839" alt="Screenshot 2023-03-17 at 3 35 01 PM" src="https://user-images.githubusercontent.com/1687902/226004215-e3dc98f2-cf0f-4f4c-b10f-cae8c0cd96c2.png">

**Comment by lachieh** [comment]
Additional research: tailwind uses `postcss-selector-parser` to escape the section [after the `group/`](https://github.com/tailwindlabs/tailwindcss/blob/acf94038261aa532d970cc13070cdcdb1b41bb0a/src/corePlugins.js#L152-L161) and postcss-selector-parser in turn uses [`cssesc`](https://github.com/mathiasbynens/cssesc) to do the dirty work. It might be an opportunity to use one of those libraries to validate the name section of the the `group/` class name.

**Comment by francoismassart** [comment]
`npm i eslint-plugin-tailwindcss@3.10.2` should fix it

**Comment by lachieh** [comment]
Thank you! 🚀

**Comment by francoismassart** [comment]
@Binero 
this is normal, it follows the [New opacity modifier syntax](https://tailwindcss.com/docs/upgrade-guide#new-opacity-modifier-syntax)
